identity-iota/src/tangle/network.rs

```diff
@@ -46,35 +56,37 @@ impl Network {
   }
 
   /// Returns the default node URL of the Tangle network.
-  pub fn default_node_url(self) -> &'static Url {
+  pub fn default_node_url(&self) -> Option<&'static Url> {
     match self {
-      Self::Mainnet => &*NODE_MAIN,
-      Self::Testnet => &*NODE_TEST,
+      Self::Mainnet => Some(&*NODE_MAIN),
+      Self::Testnet => Some(&*NODE_TEST),
+      _ => None,
     }
   }
 
   /// Returns the web explorer URL of the Tangle network.
-  pub fn explorer_url(self) -> &'static Url {
+  pub fn explorer_url(&self) -> Result<&'static Url> {
     match self {
-      Self::Mainnet => &*EXPLORER_MAIN,
-      Self::Testnet => &*EXPLORER_TEST,
+      Self::Mainnet => Ok(&*EXPLORER_MAIN),
+      Self::Testnet => Ok(&*EXPLORER_TEST),
+      _ => Err(Error::NoExplorerForPrivateTangles),
     }
   }
 
   /// Returns the web explorer URL of the given `message`.
-  pub fn message_url(&self, message_id: &str) -> Url {
-    let mut url: Url = self.explorer_url().clone();
-
+  pub fn message_url(&self, message_id: &str) -> Result<Url> {
+    let mut url = self.explorer_url()?.clone();
     // unwrap is safe, the explorer URL is always a valid base URL
     url.path_segments_mut().unwrap().push("message").push(message_id);
-    url
+    Ok(url)
   }
 
   /// Returns the name of the network as a static `str`.
-  pub const fn as_str(self) -> &'static str {
+  pub fn as_str(&self) -> Cow<'static, str> {
     match self {
-      Self::Mainnet => MAIN_NETWORK_NAME,
-      Self::Testnet => TEST_NETWORK_NAME,
+      Self::Mainnet => Cow::Borrowed(MAIN_NETWORK_NAME),
+      Self::Testnet => Cow::Borrowed(TEST_NETWORK_NAME),
+      Self::Other(network) => Cow::Owned(network.clone()),
     }
   }
```

Comment:
`as_str()` should probably be renamed `.name()` or `.network_name()` to be more clear, there are a lot of properties on a `Network`.